spec/models/weight_limits_spec.rb

```diff
@@ -91,6 +101,171 @@ def build_inventory_unit(variant, order)
       Spree::ActiveShipping::Config.set(:default_weight => 1)
     end
 
+    describe ".valid_weight_for_package?" do
+      context "with a non nil max weight" do
+        it "should return true if the max_weight is equal to zero" do
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 0).to eq true
+        end
+
+        it "should return true if the package weight is equal to the supplied max_weight" do
+          allow(package).to receive(:weight).and_return(1)
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 1).to eq true
+        end
+
+        it "should return true if the package weight is less than the supplied max_weight" do
+          allow(package).to receive(:weight).and_return(1)
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 2).to eq true
+        end
+
+        it "should return false if the package weight is greater than the supplied max_weight" do
+          allow(package).to receive(:weight).and_return(3)
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 2).to eq false
+        end
+      end
+
+      context "with a nil max weight" do
+        it "should return false if the max_weight is nil" do
+          expect(domestic_calculator.send :valid_weight_for_package?, package, nil).to eq false
+        end
+      end
+    end
+
+    describe ".country_weight_error?" do
+      context "with a package that is too heavy" do
+        it "should raise a Spree::ShippingError" do
+          allow(domestic_calculator).to receive(:valid_weight_for_package?).and_return(false)
+          expect{ domestic_calculator.send :country_weight_error?, package }.to raise_error(Spree::ShippingError)
+        end
+      end
+
+      context "with a package within the weights limit" do
+         it "should return false" do
+          allow(domestic_calculator).to receive(:valid_weight_for_package?).and_return(true)
+          expect(domestic_calculator.send :country_weight_error?, package).to be_nil
+        end
+      end
+    end
+
+    describe ".convert_package_to_weights_array" do
+      it "should use the unit_multiplier from Spree::ActiveShipping::Config" do
+        # Small package content is : 1 x Variant1, 1x Variant2, 1x Variant3
+        weights = international_calculator.send :convert_package_to_weights_array, small_package
+        active_shipping_weights = [variant1.weight, variant2.weight, variant3.weight].map do |x|
+            (x * Spree::ActiveShipping::Config[:unit_multiplier]).to_d
+        end
+        expect(weights).to match_array active_shipping_weights
+      end
+
+      context "with a package containing variants with no weight" do
+        before do
+          # It's not set globally, therefore we set a value before those tests
+          Spree::ActiveShipping::Config.set(:max_weight_per_package => 30)
```

Comment:
We should make sure that we change the config in `around` blocks and set the value back to the default after the tests dependent on the configuration are run. We had issues with this leaking into other tests and creating test-order dependent failures.